altapay.php

```diff
@@ -73,16 +78,18 @@ function altapay_add_gateway( $methods ) {
 	$terminalInfo = json_decode( get_option( 'altapay_terminals' ) );
 	if ( $terminals ) {
 		foreach ( $terminals as $terminal ) {
-			$tokenStatus  = '';
-			$terminalName = $terminal;
+			$tokenStatus   = '';
+			$subscriptions = false;
+			$terminalName  = $terminal;
 			foreach ( $terminalInfo as $term ) {
 				if ( $term->key === $terminal ) {
 					$terminalName = $term->name;
-					foreach ( $term->nature as $nature ) {
-						if ( $nature->Nature === 'CreditCard' ) {
-							$tokenStatus = 'CreditCard';
-							break;
-						}
+					$natures      = array_column( json_decode( json_encode( $term->nature ), true ), 'Nature' );
```

Comment:
@shahbaz-mehar Is this information cached in the database, or are you calling the /getTerminals endpoint?

Reply:
@emicha This information comes from the database. We are not calling /getTerminals endpoint here.
We are calling /getTerminals endpoint when the user enters the API credentials on the settings page and saves the details in the database.